docs/current/guides/snippets/build-test-publish-php/dagger.php

```diff
@@ -0,0 +1,277 @@
+<?php
+// include auto-loader
+include 'vendor/autoload.php';
+
+use GraphQL\Client;
+
+class DaggerPipeline {
+
+  private $client;
+
+  // constructor
+  public function __construct() {
+    // initialize client with
+    // endpoint from environment
+    $sessionPort = getenv('DAGGER_SESSION_PORT') or throw new Exception("DAGGER_SESSION_PORT doesn't exist");
+    $sessionToken = getenv('DAGGER_SESSION_TOKEN') or throw new Exception("DAGGER_SESSION_TOKEN doesn't exist");
+    $this->client = new Client(
+      'http://127.0.0.1:' . $sessionPort . '/query',
+      ['Authorization' => 'Basic ' . base64_encode($sessionToken . ':')]
```

Comment:
Is there a URL that we can request to check that connectivity works & credentials are correct?

Putting guards & checks as close as possible to the edge usually helps with unexpected issues that happen further away from the problem. In this case, if there is something wrong with the credentials, or the session port is incorrect, the user would find about it as close as possible to the code that handles these concerns.

Reply:
Tested this. If unable to connect for any reason, the client will throw the following error which will be caught by the handler:
```
Building test image...
Client error: `POST http://127.0.0.1:44969/query` resulted in a `401 Unauthorized` response
```

Reply:
OK.